python/hsfs/core/feature_view_engine.py

```diff
@@ -815,9 +815,27 @@ def _get_primary_keys_from_query(self, fv_query_obj):
             fv_pks.update(
                 [
                     feature.name
+                    if _join.prefix is None
+                    else _join.prefix + feature.name
                     for feature in _join.query._left_feature_group.features
                     if feature.primary
                 ]
             )
 
         return list(fv_pks)
+
+    def _get_eventtimes_from_query(self, fv_query_obj):
+        fv_events = set()
+        if fv_query_obj._left_feature_group.event_time:
+            fv_events.update([fv_query_obj._left_feature_group.event_time])
+        for _join in fv_query_obj._joins:
+            if _join.query._left_feature_group.event_time:
+                fv_events.update(
+                    [
+                        _join.query._left_feature_group.event_time
```

Comment:
should throw exception here if there are ambiguous event time column? and not wait until the query get executed.
same for primary key.